src/glue/SessionHL.cpp

```diff
@@ -78,7 +80,7 @@ std::vector<memgraph::communication::bolt::Value> TypedValueResultStreamBase::De
   std::vector<memgraph::communication::bolt::Value> decoded_values;
   decoded_values.reserve(values.size());
   for (const auto &v : values) {
-    auto maybe_value = memgraph::glue::ToBoltValue(v, *interpreter_context_->db, memgraph::storage::View::NEW);
+    auto maybe_value = memgraph::glue::ToBoltValue(v, *storage_, memgraph::storage::View::NEW);
```

Comment:
Problem, not all values in the stream will have the same storage provenance. 
TypedValue's Vertex/Edge/Path need changing + we need to guarantee access.

Reply:
This class is used to generate a temporary stream during a Pull (line 158).

Reply:
This is fine for the current implementation, but will actually be fixed when we add multi-database queries.